duckduckgo/macos-browser
# First reponder issues when switching tabs

Task/Issue URL: https://app.asana.com/0/1177771139624306/1201906946231191/f
Tech Design URL:
CC:

**Description**:
Correction of how content is switched in BrowserTabViewController.swift in order to fix selection of the first responder. 
Additionally, there is a small recatoring making code more straightforward, easier to maintain and it prevents from unnecessary code execution happening with each URL change or tab selection.

**Steps to test this PR**:
1. Switch between various content types (loaded website, new page, preferences, bookmark management page,..) and make sure address bar is focused only if new page is selected. (also make sure content is displayed correctly)
1. Verify the error view (load non-existing domain, turn of connection)


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
